src/App/Repository/EloquentRepository.php

```diff
@@ -0,0 +1,228 @@
+            ->count();
+    }
+
+    /**
+     * Return a SELECT query, optionally with preconditions.
+     * @param  Array $where optional hash of conditions
+     * @return \Illuminate\Database\Query\Builder
+     */
+    protected function selectQuery(array $where = [])
+    {
+        $query = $this->connection->table($this->getTable())
+            ->select($this->getTable() . '.*') // @todo do we need this?
+            ->where($where) // @todo do we need to handle whereIn here too?
+            ;
+
+        return $query;
+    }
+
+    /**
+     * Create a single record from input and return the created ID.
+     * @param  Array $input hash of input
+     * @return Integer
+     */
+    protected function executeInsert(array $input)
+    {
+        if (!$input) {
+            throw new RuntimeException(sprintf(
+                'Cannot create an empty record in table "%s"',
+                $this->getTable()
+            ));
+        }
+
+        return $this->connection
+            ->table($this->getTable())
+            ->insertGetId($input);
+    }
+
+    /**
+     * Update records from input with conditions and return the number affected.
+     * @param  Array $where hash of conditions
+     * @param  Array $input hash of input
+     * @return Integer
+     */
+    protected function executeUpdate(array $where, array $input)
+    {
+        if (!$where) {
+            throw new RuntimeException(sprintf(
+                'Cannot update every record in table "%s"',
+                $this->getTable()
+            ));
+        }
+
+        // Prevent overwriting created timestamp
+        // Probably not needed if `created` is set immutable in Entity
+        if (array_key_exists('created', $input)) {
+            unset($input['created']);
+        }
+
+        if (!$input) {
+            return 0; // nothing would be updated, just ignore
+        }
+
+        return $this->connection
+            ->table($this->getTable())
+            ->where($where)
+            ->update($input);
+    }
+
+    /**
+     * Delete records with conditions and return the number affected.
+     * @param  Array $where hash of conditions
+     * @return Integer
+     */
+    protected function executeDelete(array $where)
+    {
+        if (!$where) {
+            throw new RuntimeException(sprintf(
+                'Cannot delete every record in table "%s"',
+                $this->getTable()
+            ));
+        }
+
+        return $this->connection
+            ->table($this->getTable())
+            ->where($where)
+            ->delete();
+    }
+
+    /**
+     * Check if an entity with the given id exists
+     * @param  int $id
+     * @return bool
+     */
+    public function exists($id)
+    {
+        return $this->connection->table($this->getTable())
+            ->select($this->getTable() . '.*') // @todo do we need this?
+            ->where($where) // @todo do we need to handle whereIn here too?
```

Comment:
Don't need to resolve this right now... but maybe I should so it doesn't trip someone else up?